c2corg_ui/views/profile.py

```diff
@@ -0,0 +1,106 @@
+import logging
+
+from pyramid.renderers import render
+from pyramid.view import view_config
+from pyramid.httpexceptions import (
+    HTTPNotFound, HTTPInternalServerError)
+
+from c2corg_ui.views.document import Document
+from c2corg_ui.caching import cache_document_detail
+
+log = logging.getLogger(__name__)
+
+
+class Profile(Document):
+
+    _API_ROUTE = 'profiles'
+
+    @view_config(route_name='profiles_view')
+    def profile(self):
```

Comment:
I think it would be good to document how the profile page is constructed. Something like:


User profiles are a bit special, because users can mark their profile as non-public so that the profile data has to be requested with an authentication header. The request to get a profile page is made by the browser, so that the authentication header can not be set. That's why the profile page is constructed as follows:

- The browser makes a request `/profiles/123` to the UI server-side (without authentication header).
- The UI server sides makes an unauthenticated request to the API to get the available locales of the profile (`api.camptocamp.org/profiles/123/info`)
- The UI server-side returns a page containing only the user name.
- On the UI client-side a request is made to the UI server-side to get the profile data as rendered HTML (e.g. `/profiles/data/123/fr`). If the user is logged-in, the request is made authenticated.
- The UI server-side makes a request to the API to get the profile data (e.g. (`api.camptocamp.org/profiles/123/fr`)). If the request to the UI server-side was authenticated, the request to the API is also made authenticated.
- On the UI client-side the rendered HTML is inserted into the page.